stl/inc/variant

```diff
@@ -934,12 +934,11 @@ public:
         : _Mybase(in_place_index<0>) {} // value-initialize alternative 0
 
     template <class _Ty,
-        enable_if_t<sizeof...(_Types) != 0 //
-                        && !is_same_v<_Remove_cvref_t<_Ty>, variant> //
-                        && !_Is_specialization_v<_Remove_cvref_t<_Ty>, in_place_type_t> //
-                        && !_Is_in_place_index_specialization<_Remove_cvref_t<_Ty>> //
-                        && is_constructible_v<_Variant_init_type<_Ty, _Types...>, _Ty>, //
-            int> = 0>
+        enable_if_t<sizeof...(_Types) != 0 && !is_same_v<_Remove_cvref_t<_Ty>, variant>
+                        && !_Is_specialization_v<_Remove_cvref_t<_Ty>, in_place_type_t>
+                        && !_Is_in_place_index_specialization<_Remove_cvref_t<_Ty>>,
+            int>                                                                      = 0,
+        enable_if_t<is_constructible_v<_Variant_init_type<_Ty, _Types...>, _Ty>, int> = 0>

```

Comment:
I'm open to suggestions on improving the formatting of these `enable_if_t`s. Nothing I tried looked significantly better than what clang-format produces.

Reply:
Is there a way to stop it from aligning the `= 0`s? That's the only somewhat ugly thing going on here.

Reply:
Other than `clang-format off`, our normal trick is to add a blank line breaking consecutive lines with `=` so clang-format won't align. Would a weird blank line in the middle of the template head be better or worse?

Reply:
I like the idea of having a blank line here, it seems to be the least of all evils and it kind of helps with the readability of the bulky SFINAE conditions.

Reply:
I'm ambiguous, so happy to go with your preference. I'll wait for your review to make this change along with any others.

Reply:
As just discussed in Discord, let's see if we can also introduce comments here, which will both serve as formatting breaks, and explain the very subtle and rare technique being used.

It may also be worth commenting why `conjunction` isn't being used.